### PR TITLE
Add PixelShuffle-based image patching for PyTorch 2D UNet models for Patch Diffusion

### PR DESCRIPTION
**Pull request by clarencechen**
Address #2522 
Add `conv_in` and `conv_out` kernel size options for UNet2DModel
TODO: 
- Script to load weights from https://github.com/ericl122333/PatchDiffusion-Pytorch (modified from ADM codebase)
- Write tests

### CODE REVIEW

File: src/diffusers/models/unet_2d_condition.py [hunk 1]
```diff
@@ -104,6 +104,10 @@ class conditioning with `class_embed_type` equal to `None`.
             The dimension of `cond_proj` layer in timestep embedding.
         conv_in_kernel (`int`, *optional*, default to `3`): The kernel size of `conv_in` layer.
         conv_out_kernel (`int`, *optional*, default to `3`): The kernel size of `conv_out` layer.
+        patch_size (`int`, *optional*, default to `1`):
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_2d_condition.py:107]: Do we have text-conditioned checkpoints for Patch Diffusion? I'm not too keen on adding too many if-else statements if we don't have trained checkpoints for stable -diffusion

File: src/diffusers/models/unet_2d_condition.py [hunk 2]
```diff
@@ -361,6 +372,10 @@ def __init__(
             block_out_channels[0], out_channels, kernel_size=conv_out_kernel, padding=conv_out_padding
         )
 
+        self.unpatchify = None
+        if patch_size > 1:
+            self.unpatchify = nn.PixelShuffle(patch_size)
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_2d_condition.py:377]: Is PixelShuffle supported since some time?

**Review by patrickvonplaten — COMMENTED** [review]
Thanks for the PR, note that we would need a clear inference example that is enabled by those changes. Could you provide a code snippet that shows how these changes can be used in inference?

File: src/diffusers/models/unet_2d_condition.py (hunk 1 above)
Comment (clarencechen) [reply src/diffusers/models/unet_2d_condition.py:107]: Unfortunately, Patch Diffusion did not train any text-conditioned models to my knowledge, instead using dataset-specific class category embeddings. Ping @ericl122333 for more context.

**Comment by clarencechen** [comment]
Unfortunately, Patch Diffusion used dataset-specific class category embeddings to condition their trained models on benchmark datasets such as LSUN and FFHQ. I do not see any use for this PR in inference until *general-purpose* text-conditioned diffusion models are trained using this architectural choice.

**Pull request closed without merging**